docs/design/paginators.md

```diff
@@ -70,127 +69,113 @@ used in a model.
+      .collect { response ->
+         response.functions?.forEach { functionConfiguration ->
+            println(functionConfiguration.functionName) 
+         }
       }
-
-      val result = client.listFunctions(req)
-      isFirstPage = false
-      cursor = result.nextMarker
-      return result
-   }
-
-   // "functions" is the member targeted by the trait's `items` property
-   val functions: SdkAsyncIterable<FunctionConfiguration> = mapNotNull(ListFunctionsResponse::functions).flatten()
 }
 ```
 
-The shape targeted by `inputToken` is mapped to the `cursor` field. Each iteration the `cursor` is updated with the output field targeted by the `outputToken`.
-
+#### Usage - Iterating over Modeled Item
 
-### Creating a Paginator
-
-Each operation that supports pagination would receive an extension function to create a paginator. Using the model in the introduction would produce the following:
+Notice that this example is a superset of the previous.  Simply by providing a transform we are able to iterate over the 
+nested element.  A key design feature of this approach is that users may extend the abstraction as needed simply by providing
+their own flow transforms.  
 
+Additionally, the `item` specified in the model may be deeply nested and involve operations and types with many words.  Generating a
+single function that combines the operation, some word to indicate pagination, plus the nested item's name often produces
+unreadable function signatures.  In this approach, we generate more legible API by breaking the extraction of the nested
+items into a separate function.
 
 ```kotlin
-// file: Paginators.kt
-package aws.sdk.kotlin.services.lambda
-
-import aws.sdk.kotlin.services.lambda.model.FunctionConfiguration
-import aws.sdk.kotlin.services.lambda.model.ListFunctionsRequest
-
-fun LambdaClient.listFunctionsPaginated(request: ListFunctionsRequest): ListFunctionsPaginator
-    = ListFunctionsPaginator(this, request)
+lambdaClient
+  .paginateListFunctions(ListFunctionsRequest {})
+  .onFunctionConfiguration()
+  .collect { functionConfiguration ->
+      println(functionConfiguration.functionName)
+  }
 ```
 
-
-
-### Example Usage - Manual Pagination
-
-An example of driving a paginator manually and processing results:
+#### Usage - Iterating over Maps
 
 ```kotlin
-suspend fun rawPaginationExample(client: LambdaClient) {
-    val req: ListFunctionsRequest = ListFunctionsRequest{}
-    val pager: SdkAsyncIterable<FunctionConfiguration> = client.listFunctionsPaginated(req).functions
-    for (fn in pager) {
-        println(fn)
-    }
+ApiGatewayClient.fromEnvironment().use { apiGatewayClient ->
+  apiGatewayClient
+      .paginateGetUsage(GetUsageRequest { })
+      .onMapOfKeyUsages().collect { (key, value) -> // Map.Entry<String, List<List<Long>>>
+          println("${key}: ${value}")
+      }
 }
 ```
 
-
 ## Alternatives Considered
 
-
 ### Construction ALT 1 - Extension on the operation input
```

Comment:
GH won't let me comment but the example for ALT1 is still based on `SdkAsyncIterable` rather than flow.

Reply:
I left this as I wasn't sure how much of it was tied to the original design.  I've updated to mention flow.